public/js/ajax.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Do ajax request
+ * @param  {String}   method
+ * @param  {String}   url
+ * @param  {Object}   headers  `key` : `value` pair for each header
+ * @param  {Object}   data     body data
+ * @param  {Function} callback on response
+ */
+
+function ajaxRequest(method, url, headers, data, callback) {
+	let request = new XMLHttpRequest();
+	request.open(method, url, true);
+	request.onreadystatechange = function() {
+		if (request.readyState === 4) {
+			if (request.status === 200) {
```

Comment:
generalised with log of all status codes (in default), and handles specific cases of 200,201,205 only